cloud/openstack/os_ironic_inspect.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# coding: utf-8 -*-
+
+# (c) 2015-2016, Hewlett Packard Enterprise Development Company LP
+#
+# This module is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This software is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this software.  If not, see <http://www.gnu.org/licenses/>.
+
+try:
+    import shade
+    HAS_SHADE = True
+except ImportError:
+    HAS_SHADE = False
+
+from distutils.version import StrictVersion
+
+DOCUMENTATION = '''
+---
+module: os_ironic_inspect
+short_description: Explicitly triggers baremetal node introspection in ironic.
+extends_documentation_fragment: openstack
+author: "Julia Kreger (@juliakreger)"
+description:
+    - Requests Ironic to set a node into inspect state in order to collect metadata regarding the node.
+      This command may be out of band or in-band depending on the ironic driver configuration.
+      This is only possible on nodes in 'manageable' and 'available' state.
+options:
+    mac:
+      description:
+        - unique mac address that is used to attempt to identify the host.
+      required: false
+      default: None
+    uuid:
+      description:
+        - globally unique identifier (UUID) to identify the host.
+      required: false
+      default: None
+    name:
+      description:
+        - unique name identifier to identify the host in Ironic.
+      required: false
+      default: None
+    ironic_url:
+      description:
+        - If noauth mode is utilized, this is required to be set to the endpoint URL for the Ironic API.
+          Use with "auth" and "auth_type" settings set to None.
+      required: false
+      default: None
+    timeout:
+      description:
+        - A timeout in seconds to tell the role to wait for the node to complete introspection if wait is set to True.
+      required: false
+      default: 1200
+
+requirements: ["shade"]
+'''
+
+EXAMPLES = '''
+# Invoke node inspection
+- os_ironic_inspect:
+    name: "testnode1"
+'''
+
+
+def _choose_id_value(module):
+    if module.params['uuid']:
+        return module.params['uuid']
+    if module.params['name']:
+        return module.params['name']
+    return None
+
+
+def main():
+    argument_spec = openstack_full_argument_spec(
+        auth_type=dict(required=False),
+        uuid=dict(required=False),
+        name=dict(required=False),
+        mac=dict(required=False),
+        ironic_url=dict(required=False),
+        timeout=dict(default=1200, required=False),
```

Comment:
Please be specific with types (type='int' here for timeout), since a coming change to Ansible may break this.